tests/test_utils.py

```diff
@@ -59,13 +71,16 @@ def _get_dd(unique_id, serial_no, udn):
             <url>/icon2.jpg</url>
          </icon>
       </iconList>
-  <serviceList><service><serviceType>urn:schemas-upnp-org:service:AVTransport:1</serviceType><serviceId>urn:upnp-org:serviceId:AVTransport</serviceId><SCPDURL>AVTransport/scpd.xml</SCPDURL><controlURL>AVTransport/control</controlURL><eventSubURL>AVTransport/event</eventSubURL></service><service><serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType><serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId><SCPDURL>ConnectionManager/scpd.xml</SCPDURL><controlURL>ConnectionManager/control</controlURL><eventSubURL>ConnectionManager/event</eventSubURL></service><service><serviceType>urn:schemas-upnp-org:service:RenderingControl:1</serviceType><serviceId>urn:upnp-org:serviceId:RenderingControl</serviceId><SCPDURL>RenderingControl/scpd.xml</SCPDURL><controlURL>RenderingControl/control</controlURL><eventSubURL>RenderingControl/event</eventSubURL></service></serviceList><dlna:X_DLNADOC xmlns:dlna="urn:schemas-dlna-org:device-1-0">DMR-1.50</dlna:X_DLNADOC>
-<pnpx:X_hardwareId>VEN_2A2D&amp;DEV_0001&amp;SUBSYS_0001&amp;REV_01 VEN_0033&amp;DEV_0006&amp;REV_01</pnpx:X_hardwareId>
-<pnpx:X_compatibleId>MS_DigitalMediaDeviceClass_DMR_V001</pnpx:X_compatibleId>
-<pnpx:X_deviceCategory>MediaDevices</pnpx:X_deviceCategory>
-<df:X_deviceCategory>Multimedia.DMR</df:X_deviceCategory>
-<microsoft:magicPacketWakeSupported>0</microsoft:magicPacketWakeSupported>
-<microsoft:magicPacketSendSupported>1</microsoft:magicPacketSendSupported></device></root>
+      <serviceList></serviceList>
+      <dlna:X_DLNADOC xmlns:dlna="urn:schemas-dlna-org:device-1-0">DMR-1.50</dlna:X_DLNADOC>
+      <pnpx:X_hardwareId>VEN_2A2D&amp;DEV_0001&amp;SUBSYS_0001&amp;REV_01 VEN_0033&amp;DEV_0006&amp;REV_01</pnpx:X_hardwareId>
+      <pnpx:X_compatibleId>MS_DigitalMediaDeviceClass_DMR_V001</pnpx:X_compatibleId>
+      <pnpx:X_deviceCategory>MediaDevices</pnpx:X_deviceCategory>
+      <df:X_deviceCategory>Multimedia.DMR</df:X_deviceCategory>
+      <microsoft:magicPacketWakeSupported>0</microsoft:magicPacketWakeSupported>
+      <microsoft:magicPacketSendSupported>1</microsoft:magicPacketSendSupported>
+   </device>
+</root>
```

Comment:
Why did you need to change this? The original data is raw from an actual device.

Reply:
I changed this because `get_serial_number_from_host()` calls the `async_create_device` method on `UpnpFactory` which reads all of the XML files mentioned in the service meta data and I didn't have those XML files.   I assume that this XML came from your Arcam so if you could send me the three XML files (`AVTransport/scpd.xml`, `ConnectionManager/scpd.xml` and `RenderingControl/scpd.xml`) then I could update the test to include them too.

On the other hand, if you decide that we don't need `get_serial_number_from_host()` then we could remove all reference to XML altogether.